File: java/0458.0_Poor_Pigs.java
/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00% 的用户
内存消耗：35.4 MB, 在所有 Java 提交中击败了6.83% 的用户
通过测试用例：17 / 17
*/
class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int pig = 0;
        int round = minutesToTest / minutesToDie + 1;
        while (Math.pow(round, pig) < buckets) {
            pig++;
        }
        return pig;
    }
}

File: problems/0458.0_Poor_Pigs.py
'''
k轮：k+1 进制

Runtime: 34 ms, faster than 87.59% of Python3 online submissions for Poor Pigs.
Memory Usage: 13.9 MB, less than 74.45% of Python3 online submissions for Poor Pigs.
'''
class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        k = minutesToTest // minutesToDie + 1
        pig = 0
        while k ** pig < buckets:
            pig += 1
        return pig 
